### DISCUSSION TITLE
newbie needs help, instruction errors, and which data file?

### DISCUSSION THREAD

**Discussion by GorillaSapiens** [RomanIakovlev/timeshape#100] [Q&A]
instructions say "Android developers should add Timeshape to the app level gradle.build as follows"
there is no gradle.build in Android Studio Chipmunk 2021.2.1
there is however a build.gradle, i used that instead

i got an "unresolved net" error on the import net.iakovlev.timeshape.TimeZoneEngine
i resolved it by copying files from timeshape/core/src/main/java/net/iakovlev/timeshape/ into my project
i do not know if this was the corect thing to do, or if i should have done something else

instructions say "data published at https://github.com/evansiroky/timezone-boundary-builder/releases"
but it is not clear to me which of the 8 files there i need, and what i need to do with them.
looking through timeshape,i see it trying to read from asset /data.tar.zstd, but that file does not exist in the timeshape
project, nor does it exist in the timezone-boundary-builder/releases
all the files in timezone-boundary-builder seem to be zip or json, no tar files, and no tar.zstd files

so i'm stuck here, not knowing where to get asset files from or which files to use

got any clues for me?

**Comment by RomanIakovlev** [comment]
Hi, thanks for your interest in Timeshape. Regarding your problem, you should not copy the Timeshape files into your build. Instead you need to add it a dependency into your build definition. I think the README of Timeshape explains this, although I'm no Android developer myself. I think @zero0cool0 has been using Timeshape in Android, maybe they can help?

**Reply by zero0cool0** [reply]
@GorillaSapiens is correct that there is an error in the instructions, the file is called `build.gradle`. To be clear, you need to copy the snippet into your app-level `build.gradle` (the one inside the `app` directory), not the one at the top level. 

If you get an error during import, it could mean that the dependency for Timeshape has not been downloaded correctly when you ran a sync (have you run the sync?)

Also, is `mavenCentral()` added to your block in `settings.gradle` like so?
```
dependencyResolutionManagement{ 
  repositories{
        google()
        mavenCentral()
  } 
} 
```

**Reply by GorillaSapiens** [reply]
> @GorillaSapiens is correct that there is an error in the instructions, the file is called `build.gradle`. To be clear, you need to copy the snippet into your app-level `build.gradle` (the one inside the `app` directory), not the one at the top level.
> 
> If you get an error during import, it could mean that the dependency for Timeshape has not been downloaded correctly when you ran a sync (have you run the sync?)
> 
> Also, is `mavenCentral()` added to your block in `settings.gradle` like so?
> 
> ```
> dependencyResolutionManagement{ 
>   repositories{
>         google()
>         mavenCentral()
>   } 
> } 
> ```

don't know what maven is, so... probably not.  shouldn't this step be in the instructions?

also not sure how to do a sync.  spare a clue for me?

(sorry, like i said, "newbie" here)

**Reply by GorillaSapiens** [reply]
> dependencyResolutionManagement{ 
>   repositories{
>         google()
>         mavenCentral()
>   } 
> }

in settings.gradle i have
pluginManagement {
    repositories {
        gradlePluginPortal()
        google()
        mavenCentral()
    }
}
dependencyResolutionManagement {
    repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)
    repositories {
        google()
        mavenCentral()
    }
}
rootProject.name = "MyProject"
include ':app'
